app.py

```diff
@@ -100,42 +101,79 @@ def process_connection_args(rag: str, connection_args: dict) -> dict:
             connection_args["host"] = (
                 "127.0.0.1" if "HOST" not in os.environ else os.environ["HOST"]
             )
+        port = connection_args.get("port", "19530")
+        connection_args["port"] = f"{port}"
     elif rag == RAG_KG:
         if connection_args.get("host", "").lower() == "local":
             connection_args["host"] = (
                 "127.0.0.1" if "KGHOST" not in os.environ else os.environ["KGHOST"]
             )
+        port = connection_args.get("port", "7687")
+        connection_args["port"] = f"{port}"
     return connection_args
 
+def extract_and_process_params_from_json_body(
+    json: Optional[Dict], name: str, defaultVal: Optional[Any]=None,
+) -> Optional[Any]:
+    if not json:
+        return defaultVal
+    val = json.get(name, defaultVal)
+    return val
+
 @app.post("/v1/chat/completions", description="chat completions")
-def handle(
+async def handle(
     authorization: Annotated[str | None, Header()],
-    item: ChatCompletionsPostModel,
-    # request: Request, # ChatCompletionsPostModel,
+    # item: ChatCompletionsPostModel,
+    request: Request, # ChatCompletionsPostModel,
```

Comment:
Here we go back to Request due to the connection issue. If we use ChatCompletionsPostModel, biochatter-next can't communicate in this endpoint, which is a bug, and I'll fix it in next PR.